internal/controllers/clusterextension_controller.go

```diff
@@ -278,6 +286,36 @@ func (r *ClusterExtensionReconciler) installedBundle(ctx context.Context, allBun
 	return resultSet[0], nil
 }
 
+func (r *ClusterExtensionReconciler) validateBundle(bundle *catalogmetadata.Bundle) error {
+	mediaType, err := bundle.MediaType()
+	if err != nil {
+		return fmt.Errorf("unable to determine type of bundle %q: %s", bundle.Name, err)
+	}
+
+	if mediaType != catalogmetadata.MediaTypeRegistry {
+		// We only validate registry+v1 bundles
+		return nil
+	}
```

Comment:
Wondering if we actually want to have this check... Maybe we should just disallow any bundles that express dependencies?

Reply:
> Is it possible for a simple bundle to express dependencies?

Do you mean `plain+v0`? I'm not 100% sure, but I think they can have all the FBC properties.

Reply:
I thought `plain+v0` were bare manifests, so would not present with an `olm.bundle` and properties like FBC, right?

Reply:
The bundle itself is bare manifests, but the catalog will still have metadata. For example, take a look at [this demo](https://github.com/operator-framework/operator-controller/blob/main/docs/demos/coast-to-coast-q4-2023.md#create-an-initial-file-based-catalog-fbc-image).

Or am I missing something?

Reply:
You're good.   I forgot that there is still FBC associated with it, and that the v1-only `olm.bundle.mediatype` property is exercised to determine (and all my time up to now has been in v0 catalogs). 

In that case, you've talked me into it: I'd support either expanding this to remove the conditional and add tests, or handling plain+v0 bundles in a follow-up PR.

Reply:
My understanding is the same as @m1kola 

The bundle media type in this case is, IMO, unnecessary to consider and should instead assume that the input is valid FBC and therefore evaluate only the FBC contents for validation

Reply:
Although if I am misunderstanding the intention of this PR and the goal is to enforce that only `registry+v1` bundles will ever be allowed then adding a constraint to the validation to reject any other bundle media type

Reply:
Another thing to note, the way the media type property was designed was such that registry+v1 bundles would not need to be changed. If the media type property does not exist it is assumed it is a registry+v1 bundle

Reply:
After [this discussion](https://redhat-internal.slack.com/archives/C06KP34REFJ/p1714659939965849) in Slack we decided to get rid of the media type check. If any bundle installed via `ClusterExtension` be it `registry+v1` or `plain+v0` or whether else expresses any of the properties - we fail.